src/pages/CustomerInvoicesPage.js

```diff
@@ -189,49 +151,49 @@ export const CustomerInvoicesPage = ({ currentUser, navigateTo, routeName, navig
       <View style={newPageTopSectionContainer}>
         <View style={newPageTopLeftSectionContainer}>
           <SearchBar
-            onChangeText={value => dispatch(filterData(value))}
+            onChangeText={onFilterData}
             color={SUSSOL_ORANGE}
             value={searchTerm}
             style={searchBar}
           />
         </View>
-        <View style={newPageTopRightSectionContainer}>{renderNewInvoiceButton()}</View>
+        <View style={newPageTopRightSectionContainer}>
+          <NewInvoiceButton />
+        </View>
       </View>
       <DataTable
         data={data}
         extraData={dataState}
         renderRow={renderRow}
         renderHeader={renderHeader}
         keyExtractor={keyExtractor}
-        getItemLayout={memoizedGetItemLayout}
+        getItemLayout={getItemLayout}
+        columns={columns}
       />
       <BottomConfirmModal
         isOpen={hasSelection}
         questionText={modalStrings.delete_these_invoices}
-        onCancel={() => dispatch(deselectAll())}
-        onConfirm={() => dispatch(deleteTransactionsById())}
+        onCancel={onCancelRemoval}
+        onConfirm={onRemoveInvoices}
         confirmText={modalStrings.delete}
       />
-      <SelectModal
+      <DataTablePageModal
+        fullScreen={false}
         isOpen={!!modalKey}
-        options={UIDatabase.objects('Customer')}
-        placeholderText={modalStrings.start_typing_to_select_customer}
-        queryString="name BEGINSWITH[c] $0"
-        sortByString="name"
-        onSelect={name => onNewInvoice(name)}
+        modalKey={modalKey}
         onClose={() => dispatch(closeBasicModal())}
```

Comment:
`dispatch(closeBasicModal())` is used twice in the file; it could be wrapped in a local method in which case this line could be

```suggestion
        onClose={handleClose}
```

Reply:
03220bb2f1aebdaea38615cd3cf16a5095859591